### PR TITLE
feat(AbstractNavItem): expose generated props id

### LINKED ISSUE #3940: <Nav.Item> in <Nav> ignores the id prop
## Describe the bug

Can not set id attribute to ListGroupItem

## To Reproduce

Just type something like this and look into browser DevTools - ListGroup has an id, but list items doesn't
```   
	  <ListGroup id="items">
        <ListGroupItem id="item1">Pit</ListGroupItem>
        <ListGroupItem id="item2">John</ListGroupItem>
        <ListGroupItem id="item3">Ivan</ListGroupItem>
      </ListGroup>
```

Screenshot from 2019-06-22 12-39-14

## Reproducable Example

https://codesandbox.io/embed/frosty-ride-252jt

### PR DESCRIPTION
**Pull request by baktiaditya**
fix #3940

### CODE REVIEW

**Comment by taion** [comment]
I don't think this is the right approach. I think `AbstractNav` should default `getControlledId` and `getControllerId` to `null` instead of `noop` in its provided context, and that those two context values should only be used when provided.

**Comment by baktiaditya** [comment]
updated, hope this is approach is correct

**Comment by baktiaditya** [comment]
updated, hope this approach is the correct one

File: src/AbstractNavItem.js [hunk 1]
```diff
@@ -34,20 +35,26 @@ const AbstractNavItem = React.forwardRef(
...
     const navKey = makeEventKey(eventKey, props.href);
     const parentOnSelect = useContext(SelectableContext);
     const navContext = useContext(NavContext);
+    let id = _id;
 
     let isActive = active;
     if (navContext) {
       if (!props.role && navContext.role === 'tablist') props.role = 'tab';
 
+      const controllerId = navContext.getControllerId(navKey);
```
Comment (taion) [inline src/AbstractNavItem.js]: we want something like:

```js
if (navContext.getControllerId) {
  props.id = navContext.getControllerId(navKey);
}
```

with the default removed in `AbstractNav`

**Comment by atsolberg** [comment]
Any update on this PR? Not having id's is brutal for the analytics/tracking I need to do.

**Comment by taion** [comment]
My comment on https://github.com/react-bootstrap/react-bootstrap/pull/4618/files#r332227426 still stands – this PR doesn't do what it's supposed to do.

**Comment by bpas247** [comment]
Since this has been stale for quite awhile, I'm going to go ahead and close it 🙂 

@baktiaditya please feel free to re-open this if you make any new progress on incorporating @taion's feedback.

**Pull request closed without merging**